azkaban-common/src/main/java/azkaban/executor/ExecutorApiClient.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2015 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.executor;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.StatusLine;
+import org.apache.http.client.HttpResponseException;
+import org.apache.http.util.EntityUtils;
+
+import azkaban.utils.JSONUtils;
+import azkaban.utils.Pair;
+import azkaban.utils.RestfulApiClient;
+
+/** Client class that will be used to handle all Restful API calls between Executor and the host application.
+ * */
+public class ExecutorApiClient extends RestfulApiClient<Map<String, Object>> {
+
+  // hide the constructor. we what this class to be in singleton.
+  private ExecutorApiClient(){}
+
+  // international cache for the object instance.
+  private static ExecutorApiClient instance = null;
+
+  /**Singleton creator of the class.
+   * */
+  public static ExecutorApiClient getInstance() {
+    if (null == instance) {
+      synchronized (ExecutorApiClient.class) {
+        if (null == instance) {
+          instance = new ExecutorApiClient();
+        }
+      }
+    }
+    return instance;
+  }
+
+  /**Implementing the parseResponse function to return de-serialized Json object.
+   * @param response  the returned response from the HttpClient.
+   * */
+  @SuppressWarnings("unchecked")
+  @Override
+  protected Map<String, Object> parseResponse(HttpResponse response)
```

Comment:
fixed the first comment by returning the empty list, added comment in document as well. 
for the second comment as it is how the de-serializer returns result I would suggest leave as is.